code/modules/reagents/reagent_containers/food/snacks.dm

```diff
@@ -4013,20 +4069,46 @@
 	storage_slots = 3
 	w_class = W_CLASS_MEDIUM
 	food_flags = FOOD_SWEET | FOOD_ANIMAL | FOOD_LACTOSE
+	candles_state = CANDLES_UNLIT
 
 /obj/item/weapon/reagent_containers/food/snacks/sliceable/birthdaycake/New()
 	..()
 	reagents.add_reagent(NUTRIMENT, 20)
 	reagents.add_reagent(SPRINKLES, 10)
 	bitesize = 3
 
+/obj/item/weapon/reagent_containers/food/snacks/sliceable/birthdaycake/update_icon()
+	..()
+	if (candles_state == CANDLES_LIT)
+		var/image/I = image(icon,src,"[icon_state]_lit")
+		I.appearance_flags = RESET_COLOR
+		I.blend_mode = BLEND_ADD
+		if (isturf(loc))
+			I.plane = ABOVE_LIGHTING_PLANE
+		else
+			I.plane = ABOVE_HUD_PLANE // inventory
+		overlays += I
+
 /obj/item/weapon/reagent_containers/food/snacks/birthdaycakeslice
 	name = "Birthday Cake slice"
 	desc = "A slice of your birthday!"
 	icon_state = "birthdaycakeslice"
 	bitesize = 2
 	food_flags = FOOD_SWEET | FOOD_ANIMAL | FOOD_LACTOSE
 	plate_icon = "bluecustom"
+	candles_state = CANDLES_UNLIT
+
+/obj/item/weapon/reagent_containers/food/snacks/birthdaycakeslice/update_icon()
+	..()
+	if (candles_state == CANDLES_LIT)
+		var/image/I = image(icon,src,"[icon_state]_lit")
+		I.appearance_flags = RESET_COLOR
+		I.blend_mode = BLEND_ADD
+		if (isturf(loc))
```

Comment:
'I.plane = isturf(loc) ? ABOVE_LIGHTING_PLANE : ABOVE_HUD_PLANE'

Reply:
same argument as with the trash candle item.